### ISSUE TITLE
Is /var/cache/swcatalog a valid location?

### ISSUE THREAD

**Issue by Vogtinator** [ximion/appstream#583]
In openSUSE we're trying to migrate from the old app-info location to the new one, by writing XML and icons into `/var/cache/swcatalog/` instead of `/var/cache/app-info/`.

This works fine, however I did look at the documentation again and am now unsure whether this is correct.

https://www.freedesktop.org/software/appstream/docs/chap-CatalogData.html mentions:

> There are two valid locations to store AppStream XML data. /usr/share/swcatalog/xml stores all AppStream data which has been installed via software packages, while /var/lib/swcatalog/xml stores application data which was downloaded by the package manager or placed there by other tools (for example, Limba).

That does not mention `/var/cache/swcatalog/`. However, it does mention that `/var/cache/app-info` is still read so the documentation might just be wrong.

On the [YAML catalog page](https://www.freedesktop.org/software/appstream/docs/sect-AppStream-YAML.html) it does mention `/var/cache/swcatalog/`:

> Take a look at [Section 3.1.2, “File naming and location”](https://www.freedesktop.org/software/appstream/docs/chap-CatalogData.html#spec-asxml-filenaming) for AppStream XML files. While the XML data belongs into the xml subdirectory in /usr/share/swcatalog (or /var/cache/swcatalog), the YAML data is stored in the yaml subdirectory.

But not `/var/lib/swcatalog/`! Are `/var/lib/swcatalog/yaml/` and `/var/cache/swcatalog/xml/` simply undocumented features or somhow intentionally not mentioned combinations?

**Comment by ximion** [comment]
We try to avoid the `/var/cache` location, unless it is for data that can be locally regenerated without internet access.
But it is supported for sure. It's a bit annoying that there are so many locations for the data, but they all serve slightly different purposes (`/usr/share` for package-installed data, `/var/lib` for downloaded data, `/var/cache` for locally generated data (and sometimes also for downloaded stuff)).
So honestly, we should just fix the documentation to reflect that state, it won't change - and I'm sure the docs simply haven't been updated as we added more and more locations.

In other words: Using `/var/cache/swcatalog/` is fine :-)
